hal/network/ncp/cellular/cellular_hal.cpp

```diff
@@ -188,9 +203,25 @@ int cellular_device_info(CellularDevice* info, void* reserved) {
     const auto client = mgr->ncpClient();
     CHECK_TRUE(client, SYSTEM_ERROR_UNKNOWN);
 
+    CellularDeviceCached cacheRead = {};
+    int r = SystemCache::instance().get(SystemCacheKey::CELLULAR_DEVICE_INFO, (uint8_t*)&cacheRead, sizeof(cacheRead));
+
     const NcpClientLock lock(client);
-    // Ensure the modem is powered on
-    CHECK(client->on());
+
+    // If modem is off, return the cached values if present
+    if (client->ncpPowerState() != NcpPowerState::ON) {
+        if (r != sizeof(CellularDeviceCached) || cacheRead.version != CELLULAR_DEVICE_VERSION) {
+            SystemCache::instance().del(SystemCacheKey::CELLULAR_DEVICE_INFO);
```

Comment:
I dont think there are any defaults we could pick. This data is read from the modem, so we cannot guess what it may be. 
The intent here is 
1. If the modem is powered off, check if we have cached modem information to avoid turning on the modem
2. If there is no cached data, or the data is of an older format. Delete the data. The next time the modem is on and `cellular_device_info()` is called, we will update the cache then